notifications/src/main/kotlin/com/amazon/opendistroforelasticsearch/commons/notifications/model/SearchResults.kt

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.opendistroforelasticsearch.commons.notifications.model
+
+import com.amazon.opendistroforelasticsearch.notifications.util.logger
+import org.apache.lucene.search.TotalHits.Relation
+import org.apache.lucene.search.TotalHits.Relation.EQUAL_TO
+import org.apache.lucene.search.TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.io.stream.Writeable
+import org.elasticsearch.common.xcontent.ToXContent.Params
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import org.elasticsearch.search.SearchHit
+
+abstract class SearchResults<ItemClass : BaseModel> : BaseModel {
+    val startIndex: Long
+    val totalHits: Long
+    val totalHitRelation: Relation
+    val objectListFieldName: String
+    val objectList: List<ItemClass>
+
+    interface SearchHitParser {
+        fun <ItemClass> parse(it: SearchHit): ItemClass
+    }
+
+    companion object {
+        private val log by logger(SearchResults::class.java)
+        private const val START_INDEX_TAG = "startIndex"
+        private const val TOTAL_HITS_TAG = "totalHits"
+        private const val TOTAL_HIT_RELATION_TAG = "totalHitRelation"
+        private fun convertRelation(totalHitRelation: Relation): String {
+            return if (totalHitRelation == EQUAL_TO) {
+                "eq"
+            } else {
+                "gte"
+            }
+        }
+
+        private fun convertRelation(totalHitRelation: String): Relation {
+            return if (totalHitRelation == "eq") {
+                EQUAL_TO
+            } else {
+                GREATER_THAN_OR_EQUAL_TO
+            }
+        }
+    }
+
+    constructor(
+        objectListFieldName: String,
+        objectItem: ItemClass
+    ) {
+        this.startIndex = 0
+        this.totalHits = 1
```

Comment:
Creating single API for single item query and list query to avoid permission conflicts.
There are many cases when the total items can be one. may be user created only 1 item w.r.t various filters?